guides/05_chatbots/08_creating-a-website-widget-from-a-gradio-chatbot.md

```diff
@@ -0,0 +1,210 @@
+        const client = await Client.connect("https://abidlabs-gradio-playground-bot.hf.space");
+        
+        const chatToggle = document.getElementById('chat-toggle');
+        const chatContainer = document.getElementById('chat-container');
+        const closeChat = document.getElementById('close-chat');
+        const chatInput = document.getElementById('chat-input');
+        const sendButton = document.getElementById('send-message');
+        const messagesContainer = document.getElementById('chat-messages');
+    
+        chatToggle.addEventListener('click', () => {
+            chatContainer.classList.remove('hidden');
+        });
+    
+        closeChat.addEventListener('click', () => {
+            chatContainer.classList.add('hidden');
+        });
+    
+        async function sendMessage() {
+            const userMessage = chatInput.value.trim();
+            if (!userMessage) return;
+
+            appendMessage(userMessage, 'user');
+            chatInput.value = '';
+
+            try {
+                const result = await client.predict("/chat", {
+                    message: {"text": userMessage, "files": []}
+                });
+                const message = result.data[0];
+                console.log(result.data[0]);
+                const botMessage = result.data[0].join('\n');
+                appendMessage(botMessage, 'bot');
+            } catch (error) {
+                console.error('Error:', error);
+                appendMessage('Sorry, there was an error processing your request.', 'bot');
+            }
+        }
+    
+        function appendMessage(text, sender) {
+            const messageDiv = document.createElement('div');
+            messageDiv.className = `message ${sender}-message`;
+            
+            if (sender === 'bot') {
+                messageDiv.innerHTML = marked.parse(text);
+            } else {
+                messageDiv.textContent = text;
+            }
+            
+            messagesContainer.appendChild(messageDiv);
+            messagesContainer.scrollTop = messagesContainer.scrollHeight;
+        }
+    
+        sendButton.addEventListener('click', sendMessage);
+        chatInput.addEventListener('keypress', (e) => {
+            if (e.key === 'Enter') sendMessage();
+        });
+    }
+    
+    initChatWidget();
+</script>
+```
+
+### 3. That's it!
+
+Your website now has a chat widget that connects to your Gradio app! Users can click the chat button to open the widget and start interacting with your app.
+
+### Customization
+
+You can customize the appearance of the widget by modifying the CSS. Some ideas:
+- Change the colors to match your website's theme
+- Adjust the size and position of the widget
+- Add animations for opening/closing
+- Modify the message styling
+
+![](https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/gradio-guides/Screen%20Recording%202024-12-19%20at%203.32.46%E2%80%AFPM.gif)
```

Comment:
It could be nice to add a call to action here:  Share on twitter if you've built a widget and tag @gradio so we can repost. Would be cool to know if anyone is building these apps/tools.